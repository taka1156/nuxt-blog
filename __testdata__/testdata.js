const dummyImgUrl = 'http://placehold.jp/150x150.png';
const MAX_NAV_DATA = 4;
const MAX_ARTICLE_DATA = 10;
const MAX_BADGE_DATA = 5;
const MAX_TOC_DATA = 5;

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl
});

const dummyBadgeTemplate = (i, name) => ({
  id: i,
  name: name,
  img: { url: dummyImgUrl }
});

const dummyClassficationTemplate = (i, name) => ({
  id: i,
  name: name,
  img: {
    url: dummyImgUrl
  }
});

const dummyTocTemplate = i => ({
  index: i,
  escapedText: 'この文章はダミーです。',
  anchor: `anchor_${i}`
});

const dummyMarkdown = `
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
`;

const dummyArticleTemplate = i => ({
  id: i,
  ...dummyDate,
  title:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
  summary:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
  body: dummyMarkdown,
  tags: dummyTagBadges.slice(0, i),
  category: dummyBadgeTemplate(1, 'ダミーカテゴリー')
});

// dummyData
const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像'
};

const dummyDate = {
  createdAt: '2020-07-04T10:53:40.252Z',
  updatedAt: '2020-07-08T15:15:07.668Z'
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyTagBadges = dummyFactory(MAX_BADGE_DATA, i =>
  dummyBadgeTemplate(i, 'ダミータグ')
);

const dummyClassificationCategory = dummyFactory(MAX_BADGE_DATA, i =>
  dummyClassficationTemplate(i, 'ダミーカテゴリー')
);

const dummyClassificationTag = dummyFactory(MAX_BADGE_DATA, i =>
  dummyClassficationTemplate(i, 'ダミータグ')
);

const dummyArticles = dummyFactory(MAX_ARTICLE_DATA, i => dummyArticleTemplate(i));

const dummyCategoryBadge = {
  badgeType: 'category',
  badge: dummyBadgeTemplate(1, 'ダミーカテゴリー')
};

const dummyTagBadge = {
  badgeType: 'tag',
  badge: dummyBadgeTemplate(1, 'ダミータグ')
};

const dummyPagination = {
  currentPage: 1,
  maxPage: Math.ceil(dummyArticles.length / 5)
};

const dummyTocs = dummyFactory(MAX_TOC_DATA, i => dummyTocTemplate(i));

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyCategoryBadge,
  dummyTagBadge,
  dummyTagBadges,
  dummyDate,
  dummyArticles,
  dummyPagination,
  dummyClassificationCategory,
  dummyClassificationTag,
  dummyLogo,
  dummyTocs,
  dummyRoutes
};
