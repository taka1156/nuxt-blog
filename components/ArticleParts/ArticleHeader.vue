<template>
  <div class="ArticleHeader">
    <div class="m-3">
      <article-category :category="article.category" />
      <article-date
        :created-at="article.createdAt"
        :updated-at="article.updatedAt"
      />
      <div class="border" />
      <!--一覧表示の時はh3、記事表示はh1-->
      <slot name="title"></slot>
      <!--一覧表示の時はsummaryの表示-->
      <div v-if="article.summary != null">
        <p class="text-left">{{ article.summary }}</p>
        <div class="border"></div>
      </div>
      <article-tag :tags="article.tags" />
    </div>
  </div>
</template>

<script>
import ArticleCategory from './ArticleCategory';
import ArticleTag from './ArticleTag';
import ArticleDate from './ArticleDate';

export default {
  name: 'ArticleHeader',
  components: {
    'article-category': ArticleCategory,
    'article-tag': ArticleTag,
    'article-date': ArticleDate
  },
  props: {
    article: {
      type: Object,
      default: () => {},
      required: true
    }
  }
};
</script>

<style></style>
