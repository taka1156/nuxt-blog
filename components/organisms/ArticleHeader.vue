<template>
  <div>
    <div class="article-header">
      <article-category :category="article.category" />
      <article-date
        :created-at="article.createdAt"
        :updated-at="article.updatedAt"
      />
      <div class="line" />
      <h1 class="article-header__title">
        {{ article.title }}
      </h1>
      <article-tag :tags="article.tags" />
    </div>
  </div>
</template>

<script>
import ArticleCategory from '../molecules/ArticleCategory';
import ArticleTag from '../molecules/ArticleTag';
import ArticleDate from '../molecules/ArticleDate';

export default {
  name: 'ArticleHeader',
  components: {
    'article-category': ArticleCategory,
    'article-tag': ArticleTag,
    'article-date': ArticleDate
  },
  props: {
    article: {
      type: Object,
      default: () => {},
      required: true
    }
  }
};
</script>

<style scoped>
.article-header {
  padding: 10px;
  background-color: whitesmoke;
  border: 1px solid lightgray;
}

.article-header__title {
  font-size: 1.75rem;
  text-align: left;
  word-break: break-word;
  overflow-wrap: break-word;
  color: rgba(40, 167, 69, 0.9);
}
</style>
