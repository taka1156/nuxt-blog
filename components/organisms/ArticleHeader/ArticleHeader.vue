<template>
  <div>
    <div class="article-header">
      <article-category :category="article.category" />
      <article-date
        :created-at="article.createdAt"
        :updated-at="article.updatedAt"
      />
      <base-heading :h-lv="1">
        {{ article.title }}
      </base-heading>
      <article-tag :tags="article.tags" />
    </div>
  </div>
</template>

<script>
import BaseHeading from '../../atoms/BaseHeading/BaseHeading';
import ArticleCategory from '../ArticleCategory/ArticleCategory';
import ArticleTag from '../ArticleTag/ArticleTag';
import ArticleDate from '../../molecules/ArticleDate/ArticleDate';

export default {
  name: 'ArticleHeader',
  components: {
    BaseHeading,
    ArticleCategory,
    ArticleTag,
    ArticleDate
  },
  props: {
    /**
     * 記事の情報
     */
    article: {
      type: Object,
      required: true
    }
  }
};
</script>

<style scoped>
::v-deep .base-heading1--extend {
  padding: 10px 0 0;
  margin-left: 5px;
  font-size: 1.75rem;
  color: rgb(40 167 69);
  text-align: left;
  border-top: 2.5px solid lightgray;
}

.article-header {
  padding: 10px;
  background-color: whitesmoke;
  border: 1px solid lightgray;
}
</style>
