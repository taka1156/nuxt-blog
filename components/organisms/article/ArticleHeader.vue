<template>
  <div>
    <div class="article-header">
      <article-category
        :category="article.category"
        @classfication-jump="classficationJump"
      />
      <article-date
        :created-at="article.createdAt"
        :updated-at="article.updatedAt"
      />
      <div class="line" />
      <h1 class="article-header__title">
        {{ article.title }}
      </h1>
      <article-tag :tags="article.tags" @classfication-jump="classficationJump" />
    </div>
  </div>
</template>

<script>
import ArticleCategory from '../../molecules/article/ArticleCategory';
import ArticleTag from '../../molecules/article/ArticleTag';
import ArticleDate from '../../molecules/article/ArticleDate';

export default {
  name: 'ArticleHeader',
  components: {
    'article-category': ArticleCategory,
    'article-tag': ArticleTag,
    'article-date': ArticleDate
  },
  props: {
    article: {
      type: Object,
      default: () => {},
      required: true
    }
  },
  methods: {
    classficationJump({ path, id }) {
      this.$router.push({
        name: `${path}-id`,
        params: { id: id }
      });
    }
  }
};
</script>

<style scoped>
.article-header {
  padding: 10px;
  background-color: whitesmoke;
  border: 1px solid lightgray;
}

.article-header__title {
  font-size: 1.75rem;
  text-align: left;
  word-break: break-word;
  overflow-wrap: break-word;
  color: rgba(40, 167, 69, 0.9);
}
</style>
