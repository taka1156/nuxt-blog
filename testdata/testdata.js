const dummyImg = {
  imgUrl: 'http://placehold.jp/150x150.png',
  imgAlt: 'ダミーの画像'
};

const dummyCategoryBadge = {
  badgeType: 'category',
  badge: {
    id: 1,
    name: 'ダミーカテゴリー',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
};

const dummyTagBadge = {
  badgeType: 'tag',
  badge: {
    id: 1,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
};

const dummyTagBadges = [
  {
    id: 1,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 2,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 3,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 4,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
];

const dummyDate = {
  createdAt: '2020-07-04T10:53:40.252Z',
  updatedAt: '2020-07-08T15:15:07.668Z'
};

const dummyMarkdown = `
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
`;

const dummyArticles = [
  {
    id: '1pbb3xp8x',
    createdAt: '2020-09-13T14:28:35.489Z',
    updatedAt: '2020-09-13T16:01:29.403Z',
    publishedAt: '2020-09-13T14:28:35.489Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'xizlxccff',
    createdAt: '2020-09-07T04:35:28.817Z',
    updatedAt: '2020-09-07T12:59:56.511Z',
    publishedAt: '2020-09-07T04:35:28.817Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: 'aamcmuhtf',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: 'xghqpsmpp',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'pgxhtaaa9z',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'b-zlnpffc',
    createdAt: '2020-07-04T12:24:08.429Z',
    updatedAt: '2020-09-07T13:00:12.392Z',
    publishedAt: '2020-07-04T12:24:08.429Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'svdq1mmtc',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  }
];

const dummyPagination = {
  currentPage: 1,
  maxPage: Math.ceil(dummyArticles.length / 5)
};

const dummyClassfication = [
  {
    id: 'xfrbr-upq',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'tlx_v85px',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'xghqpsmpp',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'dhyw5n7iw',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  }
];

const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'ダミー',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png'
  }
];

export {
  dummyImg,
  dummyCategoryBadge,
  dummyTagBadge,
  dummyTagBadges,
  dummyDate,
  dummyArticles,
  dummyPagination,
  dummyClassfication,
  dummyLogo,
  dummyRoutes
};
