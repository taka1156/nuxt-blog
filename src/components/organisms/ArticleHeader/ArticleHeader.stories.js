import ArticleHeader from './ArticleHeader.vue';
import { dummyArticles } from '__testdata__/testdata.js';

export default {
  title: 'Organisms/ArticleHeader',
  component: ArticleHeader
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ArticleHeader },
  template: '<article-header v-bind="$props" />'
});

export const Default = Template.bind({});

Default.args = { article: dummyArticles[0] };
