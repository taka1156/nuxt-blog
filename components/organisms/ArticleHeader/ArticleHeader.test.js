import { shallowMount } from '@vue/test-utils';
import ArticleHeader from './ArticleHeader.vue';
import { dummyArticles } from '@/__testdata__/testdata.js';

describe('ArticleHeader', () => {
  const articleHeader = propsData =>
    shallowMount(ArticleHeader, {
      propsData: {
        ...propsData
      }
    });

  it('ArticleHeader初期値: article', () => {
    const wrapper = articleHeader({ article: dummyArticles[0] });
    expect(wrapper.vm.$options.props.article.required).toBe(true);
    expect(wrapper.vm.article).toBe(dummyArticles[0]);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
