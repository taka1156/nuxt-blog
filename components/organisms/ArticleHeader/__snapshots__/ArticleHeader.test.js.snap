// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeader ArticleHeader初期値: article 1`] = `
"<div>
  <div class=\\"article-header\\">
    <article-category-stub category=\\"[object Object]\\"></article-category-stub>
    <article-date-stub createdat=\\"2020-07-04T10:53:40.252Z\\" updatedat=\\"2020-07-08T15:15:07.668Z\\"></article-date-stub>
    <base-heading-stub hlv=\\"1\\">
      この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています
    </base-heading-stub>
    <article-tag-stub tags=\\"[object Object]\\"></article-tag-stub>
  </div>
</div>"
`;
